### PR TITLE
Fix VectorContinousCallback

### PR DESCRIPTION
**Pull request by kanav99**
Reviews @ChrisRackauckas, if you want to dig into Callback Code again :stuck_out_tongue:

### CODE REVIEW

File: src/callbacks.jl [hunk 1]
```diff
@@ -475,7 +475,7 @@ function find_callback_time(integrator,callback::VectorContinuousCallback,counte
         bottom_θ = typeof(integrator.t)(0)
       end
       if callback.rootfind && !isdiscrete(integrator.alg)
-        minΘ = 1.0
+        minΘ = nextfloat(1.0)
```
Comment (ChrisRackauckas) [inline src/callbacks.jl]: what is this actually doing? And why only VectorContinuousCallback?

Comment (kanav99) [reply src/callbacks.jl]: For VectorContinuousCallback, we can have multiple events in same interval (in the same interpolation interval too), so we have to find the exact time where the event happened and then find the minimum time. That's what `min_theta` is about. We find the minimum time where an event happens. We started `min_theta` with 1.0. But in the case when the only event that happens is at 1.0, the `theta < min_theta` branch will be missed. So I used the next float to solve this issue.
But before merging this, I want to make sure we haven't already handled this case somewhere else. If we have already handled this, this problem might be of a larger issue.

Comment (devmotion) [reply src/callbacks.jl]: Why not just set it to `Inf`?

Comment (ChrisRackauckas) [reply src/callbacks.jl]: Inf is Float64. We should probably make this match `nextfloat(typeof(integrator.t)(1))` (and this might need a comment)

**Comment by ChrisRackauckas** [comment]
@kanav99 can you confirm that this works with the new downstream tests?

File: src/callbacks.jl (hunk 1 above)
Comment (ChrisRackauckas) [reply src/callbacks.jl]: yeah that might be better

**Comment by kanav99** [comment]
Sorry I was away today. I get some type conversion errors in CallbackCache, might be due to recent changes. Fixing them

**Comment by kanav99** [comment]
Done https://github.com/JuliaDiffEq/OrdinaryDiffEq.jl/pull/938/commits/ac05f7e6a113935542d4c3ce67df069c9f58968e.
I will run full tests now, repeat tests work now.

**Comment by kanav99** [comment]
OrdinaryDiffEq tests pass, DiffEqBase is failing on master itself

**Comment by kanav99** [comment]
Why are Downstream tests only failing for Julia 1?

**Comment by devmotion** [comment]
I'm not sure, I've seen this before in https://github.com/JuliaDiffEq/DiffEqBase.jl/pull/339 but assumed that these downstream test errors were some spurious artefacts since the Travis CI push tests still passed (only the PR tests failed). The test failure of the Core test group on Julia nightly is not an artefact, though; the downstream test failures on Julia nightly should be fixed by https://github.com/JuliaMath/Quadmath.jl/pull/40.

**Pull request merged**